### ISSUE TITLE
sagemath:latest appears to be on Ubuntu 15.10 (Wily Werewolf), instead of 16.04 (Xenial)

### ISSUE THREAD

**Issue by cemulate** [sagemath/docker-images#31]
**Edit**: Found the real problem; original symptom is described below

It appears that `sagemath/sagemath-base` is based off of `ubuntu:xenial`.
However, the image `sagemath/sagemath:latest` appears to have OS version Ubuntu 15.10 (wily):

```
> docker run -it --entrypoint=bash sagemath/sagemath:latest
sage@364b85a68ee4:~$ cat /etc/os-release
NAME="Ubuntu"
VERSION="15.10 (Wily Werewolf)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 15.10"
VERSION_ID="15.10"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
sage@364b85a68ee4:~$
```
This is causing problems with `apt-get update` as described below. Is it possible that `sagemath-base` got changed but the `sagemath` image never got rebuilt?

This is not the case when using `sagemath/sagemath:8.1` -- it's on Ubuntu 16.04 as expected.

Again, I'm pretty new to docker so if this phenomenon is normal or expected, let me know and feel free to close!

-----------------------

**Original Problem**: Apologies if this isn't the right place for this - I'm very new to docker so feel free to close if this is an inappropriate thread.

I'm trying to make a Docker image based off `sagemath/sagemath`.
With dockerfile:

```
FROM sagemath/sagemath
USER root
RUN apt-get update
```

The `apt-get update` fails with

```
Ign http://archive.ubuntu.com wily InRelease
Ign http://archive.ubuntu.com wily-updates InRelease
Ign http://archive.ubuntu.com wily-security InRelease

...

Err http://archive.ubuntu.com wily/main Sources
  404  Not Found [IP: 91.189.88.162 80]
Err http://archive.ubuntu.com wily/restricted Sources
  404  Not Found [IP: 91.189.88.162 80]
Err http://archive.ubuntu.com wily/universe Sources
  404  Not Found [IP: 91.189.88.162 80]

...

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/wily/main/source/Sources  404  Not Found [IP: 91.189.88.162 80]

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/wily/restricted/source/Sources  404  Not Found [IP: 91.189.88.162 80]

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/wily/universe/source/Sources  404  Not Found [IP: 91.189.88.162 80]

...

E: Some index files failed to download. They have been ignored, or old ones used instead.
```

The internet suggests that this is a DNS issue, but I see that `sagemath-base` begins with `FROM ubuntu:xenial`, and indeed, when I start there:

```
FROM ubuntu:xenial
RUN apt-get update
```

Everything goes smoothly as expected, which leads me to believe that something about the `sagemath/sagemath` image is causing `apt-get update` to fail.
It seems like it's trying to use repositories for Ubuntu `wily`, but `sagemath/sagemath` is based on `ubuntu:xenial`... ?

Again, I'm brand new to docker so let me know if I'm missing something obvious!

**Comment by embray** [comment]
Hi @cemulate , thanks for the detailed analysis of the problem.  It looks to me that the `latest` tag for `sagemath/sagemath` just didn't get updated for some reason when I pushed the image for `sagemath/sagemath:8.1`--normally that is part of the process.

While I'm at it I'll also rebuild the 8.0 image on top of xenial.  The base image was only changed in #27 but I think the 8.0 image is still from wily.

**Comment by embray** [comment]
I went ahead an updated the `latest` tag to point to `8.1`.   The problem with Docker (well, one of them) is that the "latest" tag is not always necessarily *really* the latest.  There's nothing special about "latest" except that it's the default tag used when pulling an image if you don't specify a tag.  So it doesn't actually necessarily point to the latest version of an image unless its maintainer actually remembers to update which version it points to.  

That is now fixed so it should resolve the issue, though I'm also rebuilding the 8.0 image to base it on xenial.
